pytest_mitmproxy.py

```diff
@@ -1 +1,51 @@
 # -*- coding: utf-8 -*-
+import re
+import os
+import shutil
+import pytest
+from contextlib import contextmanager
+from subprocess import Popen, PIPE, SubprocessError
+
+# clean the dumps each run
+if os.path.isdir("traffic_dumps"):
+    shutil.rmtree("traffic_dumps")
+
+
+@contextmanager
+def new_proxy(name: str, scope: str, flt: str=None):
+    ''' Start a new mitmdump process and
+    return the address for clients to connect
+
+    The captured content will be written to
+    ./<dump_dir>/<scope>/<name>
+    '''
+    dump_dir = "traffic_dumps"
+
+    os.makedirs("{}/{}".format(dump_dir, scope), mode=0o755, exist_ok=True)
+    dump_file = "{}/{}/{}".format(dump_dir, scope, name)
+
+    args = ["mitmdump", "--port=0", "--wfile={}".format(dump_file)]
+    args.append(flt) if flt else None
+
+    ps = Popen(
+        args,
+        stdout=PIPE,
+        stderr=PIPE,
+        universal_newlines=True,
+        bufsize=1
+    )
+    if ps.poll() is not None:
+        raise SubprocessError("mitmdump was unable to run")
+    status = ps.stdout.readline()
```

Comment:
Well, I don't know if it does. Maybe it doesn't. But if it does continously write to either stdout or stderr for *any* reason (say it only starts writing to stderr on a configuration error), then this could be a potential problem.

Nothing like having a stuck binary.